In [1]:
! pip install keras
!pip install tensorflow

In [2]:
import pandas as pd
import numpy as np

import tensorflow as tf
from tensorflow.keras.preprocessing.text import Tokenizer
from tensorflow.keras.preprocessing.sequence import pad_sequences
from tensorflow.keras.models import Sequential
from tensorflow.keras.callbacks import Callback
from tqdm import tqdm
from keras.layers import Dense
from keras.layers import Dropout
from sklearn.preprocessing import StandardScaler
from keras.callbacks import EarlyStopping, ModelCheckpoint
from tqdm.keras import TqdmCallback


# 1 Hour

In [5]:
# We take the document when we have all our data normalized:
df_1H = pd.read_csv('/content/sample_data/Full_data_1H.csv')

In [8]:
df_1H.head()

Date_1H                                           datetime  \
0  2022-01-01 01:00:00  ['2022-01-01 00:00:00', '2022-01-01 00:00:00',...   
1  2022-01-01 02:00:00  ['2022-01-01 01:00:00', '2022-01-01 01:00:00',...   
2  2022-01-01 03:00:00  ['2022-01-01 02:00:00', '2022-01-01 02:00:00',...   
3  2022-01-01 04:00:00  ['2022-01-01 03:00:00', '2022-01-01 03:00:00',...   
4  2022-01-01 05:00:00  ['2022-01-01 04:00:00', '2022-01-01 04:00:00',...   

                                                text  \
0  Current Price of Bitcoin:\n$46320 (-1.85%)\n\n...   
1  Today's Cryptocurrency Fear And Greed Index: 2...   
2  Huobi Singapore would like to wish you a Happy...   
3  1 ARS is 10.26 satoshis right now.\n\n1 satosh...   
4  Buy Bitcoin $BTC @ 46813.2\nAn online trader h...   

                                            username  kk08_pos_score  \
0  CryptoNerdApp\nTrendSpider\nMadStudentScie1\nW...        0.755714   
1  FearAndGreedBot\nOfficialOverbit\nBTC_AI_bot\n...        0.775810   
2  HuobiSg\nAmazonCash4Cars\nBTC_AI_bot\nCalvinAy...        0.773046   
3  SatoshiARS_bot\nbitcoinpartyall\ncryptopediaK7...        0.760779   
4  BTC_AI_bot\nnewscomauHQ\ncryptozone007\nCarbon...        0.771374   

   kk08_score_diff  Elkulako_label  Sentiment_vader  pred  analysis  ...  \
0         0.511428        0.364813         0.161558    34        72  ...   
1         0.551621        0.343045         0.174069    30        29  ...   
2         0.546093        0.427992         0.181803    23        24  ...   
3         0.521558        0.325186         0.175078    26        20  ...   
4         0.542747        0.377858         0.152699    26        19  ...   

   collapse  ATH  grow  rise  fall                 date      open     close  \
0         2   28    10    20    65  2022-01-01 01:00:00  46656.14  46778.14   
1         0   14    13    13    30  2022-01-01 02:00:00  46778.14  46811.77   
2         1   20    10    11    24  2022-01-01 03:00:00  46811.77  46813.20   
3         1   24    13    12    22  2022-01-01 04:00:00  46813.21  46711.05   
4         1   22     7    12    16  2022-01-01 05:00:00  46711.05  47192.55   

     Return  Return_sign  
0  0.002615            1  
1  0.000719            1  
2  0.000031            1  
3 -0.002182            0  
4  0.010308            1  

[5 rows x 31 columns]

In [7]:
df_1H.columns

Index(['Date_1H', 'datetime', 'text', 'username', 'kk08_pos_score',
       'kk08_score_diff', 'Elkulako_label', 'Sentiment_vader', 'pred',
       'analysis', 'short', 'long', 'buy', 'sell', 'bull', 'bear', 'bankrupt',
       'fraud', 'cbdc', 'record', 'crisis', 'collapse', 'ATH', 'grow', 'rise',
       'fall', 'date', 'open', 'close', 'Return', 'Return_sign'],
      dtype='object')

In [12]:
# I just colect the information i will need to use to train the NN for each of the sentiment results:
df_1H_Vader = df_1H[['Sentiment_vader','Return_sign']]
df_1H_kk08 = df_1H[['kk08_pos_score','Return_sign']]
df_1H_Elkulako = df_1H[['Elkulako_label','Return_sign']]


We define the model we want to use

In [14]:
model = Sequential([
    Dense(32, activation='relu', input_shape=(1,)),
    Dropout(0.3),
    Dense(16, activation='relu'),
    Dropout(0.3),
    Dense(8, activation='relu'),
    Dense(1, activation='sigmoid')
])

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


In [15]:
model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])



In [ ]:
# Define callbacks to stop and dont overfit the model
early_stopping = EarlyStopping(monitor='val_loss', patience=10, restore_best_weights=True)
model_checkpoint = ModelCheckpoint('best_model.keras', save_best_only=True, monitor='val_loss')

## KK08

In [17]:
df_1H_kk08.columns

Index(['kk08_pos_score', 'Return_sign'], dtype='object')

In [24]:
X_kk08 = df_1H_kk08 ['kk08_pos_score']
Y_kk08 = df_1H_kk08 ['Return_sign']
#Y_kk08 = np.array(df_1H_kk08 ['Return_sign'])

In [23]:
train_share = 0.85
df_x_1_train = X_kk08.iloc[:int(len(X_kk08)*train_share)].reset_index(drop=True)
df_y_train = Y_kk08.iloc[:int(len(Y_kk08)*train_share)].reset_index(drop=True)

df_x_1_test = X_kk08.iloc[int(len(X_kk08)*train_share):].reset_index(drop=True)
df_y_test = Y_kk08.iloc[int(len(X_kk08)*train_share):].reset_index(drop=True)

In [78]:
df_x_1_train.shape

(632,)

In [25]:
# Checking that class distribution is more or less balance in both dataset.
print("Training set class distribution:")
print(pd.Series(df_y_train).value_counts())
print("Test set class distribution:")
print(pd.Series(df_y_test).value_counts())

Training set class distribution:
Return_sign
1    322
0    310
Name: count, dtype: int64
Test set class distribution:
Return_sign
1    59
0    53
Name: count, dtype: int64


In [29]:
train_share = 0.90
split_index = int(train_share * len(df_x_1_train))
X_train, X_valid = df_x_1_train[:split_index], df_x_1_train[split_index:]
y_train, y_valid = df_y_train[:split_index], df_y_train[split_index:]

In [30]:
history = model.fit(
    df_x_1_train, df_y_train,
    epochs=200,
    batch_size=32,
    validation_data=(X_valid,  y_valid),
    callbacks=[early_stopping, model_checkpoint, TqdmCallback(verbose=1)],
    verbose=0
)

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

In [32]:
# Evaluar en el conjunto de validación o de prueba
test_loss, test_accuracy = model.evaluate(df_x_1_test, df_y_test)
print(f"Test Loss: {test_loss}")
print(f"Test Accuracy: {test_accuracy}")

4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4847 - loss: 0.6937  
Test Loss: 0.6924704909324646
Test Accuracy: 0.5267857313156128


In [35]:
from sklearn.metrics import roc_auc_score

# Evaluate on the test set
test_loss, test_accuracy = model.evaluate(df_x_1_test, df_y_test)
print(f"Test Loss: {test_loss}")
print(f"Test Accuracy: {test_accuracy}")

# Predict probabilities for the positive class
y_pred_probs = model.predict(df_x_1_test)

# Use df_y_test directly as y_true if it's already binary (0 or 1)
y_true = df_y_test  # Assuming df_y_test is not one-hot encoded

# Compute the ROC AUC score
roc_auc = roc_auc_score(y_true, y_pred_probs)
print(f"ROC AUC: {roc_auc}")


4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4847 - loss: 0.6937 
Test Loss: 0.6924704909324646
Test Accuracy: 0.5267857313156128
4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step 
ROC AUC: 0.5062360089542692


In [15]:
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.optimizers import Adam
from tensorflow.keras.callbacks import EarlyStopping
import numpy as np

In [37]:
# Since the first result was not completely satisfactory, we try to improve the solution proving adding or remove layers of a NN to se how we can improve the NN:
def create_model(num_layers=6, neurons_per_layer=48, input_shape=(10,), output_units=1, activation='relu'):
    model = Sequential()
    model.add(Dense(neurons_per_layer, activation=activation, input_shape=input_shape))

    # Agregar las capas ocultas especificadas
    for _ in range(num_layers - 1):
        model.add(Dense(neurons_per_layer, activation=activation))

    # Capa de salida
    model.add(Dense(output_units, activation='sigmoid'))  # Cambia 'sigmoid' a 'softmax' si tienes una clasificación multiclase

    # Compilar el modelo
    model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
    return model




# Definir parámetros de la red
num_layers_list = [2, 3, 4,6,8]  # Diferentes números de capas a probar
neurons_per_layer_list = [16, 32, 64,128]  # Diferentes números de neuronas en cada capa

# Probar diferentes configuraciones de capas y neuronas
for num_layers in num_layers_list:
    for neurons_per_layer in neurons_per_layer_list:
        print(f"\nModel trained with {num_layers} hidden layers andy {neurons_per_layer} neurons per layer")

        # Crear el modelo
        model = create_model(num_layers=num_layers, neurons_per_layer=neurons_per_layer, input_shape=(1,))

        # Definir early stopping para evitar el sobreajuste
        early_stopping = EarlyStopping(monitor='val_loss', patience=5, restore_best_weights=True)

        # Entrenar el modelo
        history = model.fit(df_x_1_train, df_y_train, validation_split=0.2, epochs=200, batch_size=16, callbacks=[early_stopping,TqdmCallback(verbose=1)], verbose=1)

        # Evaluar el modelo
        loss, accuracy = model.evaluate(df_x_1_test, df_y_test, verbose=0)
        print(f"Model with {num_layers} layers y {neurons_per_layer} neurons -> Loss: {loss:.4f}, accuracy: {accuracy:.4f}")

        # Predict probabilities for the positive class
        y_pred_probs = model.predict(df_x_1_test)

        # Use df_y_test directly as y_true if it's already binary (0 or 1)
        y_true = df_y_test  # Assuming df_y_test is not one-hot encoded

        # Compute the ROC AUC score
        roc_auc = roc_auc_score(y_true, y_pred_probs)
        print(f"ROC AUC: {roc_auc}")




Entrenando modelo con 2 capas ocultas y 16 neuronas por capa


/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 4s 27ms/step - accuracy: 0.5135 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6969
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.4944 - loss: 0.6938 - val_accuracy: 0.4409 - val_loss: 0.6943
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.4881 - loss: 0.6937 - val_accuracy: 0.4409 - val_loss: 0.6936
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 11ms/step - accuracy: 0.5159 - loss: 0.6929 - val_accuracy: 0.4409 - val_loss: 0.6955
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 11ms/step - accuracy: 0.4632 - loss: 0.6950 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5075 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6947
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5016 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6941
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4967 - loss: 0.6933 - val_accuracy: 0.4409 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 2s 10ms/step - accuracy: 0.5214 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6977
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5077 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6950
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5074 - loss: 0.6927 - val_accuracy: 0.5118 - val_loss: 0.6931
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5017 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6945
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.5035 - loss: 0.6933 - val_accuracy: 0.5591 - val_loss: 0.6927
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4811 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6946
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.4861 - loss: 0.6934 - val_accuracy: 0.5591 - val_loss: 0.6925
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4820 - loss: 0.6933 - val_accuracy: 0.4488 - 

4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 34ms/step


/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


ROC AUC: 0.49376399104573077

Entrenando modelo con 2 capas ocultas y 64 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 2s 11ms/step - accuracy: 0.4843 - loss: 0.6937 - val_accuracy: 0.4409 - val_loss: 0.6952
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5176 - loss: 0.6928 - val_accuracy: 0.5591 - val_loss: 0.6930
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4893 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6940
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5406 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5246 - loss: 0.6932 - val_accuracy: 0.5591 - val_loss: 0.6921
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4912 - loss: 0.6933 - val_accuracy: 0.5591 - val_loss: 0.6930
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5021 - loss: 0.6933 - val_accuracy: 0.5591 - val_loss: 0.6928
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5011 - loss: 0.6934 - val_accuracy: 0.4409 - 

Model with 2 layers y 64 neurons -> Loss: 0.6927, accuracy: 0.5268
4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 24ms/step
ROC AUC: 0.5060761112887752

Entrenando modelo con 2 capas ocultas y 128 neuronas por capa


/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 2s 9ms/step - accuracy: 0.5087 - loss: 0.6961 - val_accuracy: 0.4409 - val_loss: 0.6987
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5143 - loss: 0.6923 - val_accuracy: 0.5591 - val_loss: 0.6923
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.4727 - loss: 0.6942 - val_accuracy: 0.5591 - val_loss: 0.6920
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5205 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6935
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4561 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6967
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5134 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6955
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4451 - loss: 0.6962 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4807 - loss: 0.6944 - val_accuracy: 0.4409 - v

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 19ms/step - accuracy: 0.5073 - loss: 0.6961 - val_accuracy: 0.4409 - val_loss: 0.6993
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.4798 - loss: 0.6950 - val_accuracy: 0.4488 - val_loss: 0.6937
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.5059 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6952
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 15ms/step - accuracy: 0.4876 - loss: 0.6936 - val_accuracy: 0.4488 - val_loss: 0.6936
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 10ms/step - accuracy: 0.5173 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6941
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5002 - loss: 0.6930 - val_accuracy: 0.5591 - val_loss: 0.6923
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 11ms/step - accuracy: 0.4540 - loss: 0.6936 - val_accuracy: 0.5197 - val_loss: 0.6931
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.4891 - loss: 0.6933 - val_accuracy: 0.5591

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 2s 10ms/step - accuracy: 0.4748 - loss: 0.6971 - val_accuracy: 0.4409 - val_loss: 0.6955
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5413 - loss: 0.6919 - val_accuracy: 0.4409 - val_loss: 0.6960
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5043 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6948
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5172 - loss: 0.6930 - val_accuracy: 0.4409 - val_loss: 0.6950
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5080 - loss: 0.6929 - val_accuracy: 0.4409 - val_loss: 0.6934
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5151 - loss: 0.6932 - val_accuracy: 0.5118 - val_loss: 0.6931
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5017 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6939
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5206 - loss: 0.6929 - val_accuracy: 0.4409 - 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 2s 10ms/step - accuracy: 0.4648 - loss: 0.6948 - val_accuracy: 0.4409 - val_loss: 0.6947
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4339 - loss: 0.6946 - val_accuracy: 0.5276 - val_loss: 0.6932
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5163 - loss: 0.6936 - val_accuracy: 0.4567 - val_loss: 0.6931
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4748 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6944
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5045 - loss: 0.6931 - val_accuracy: 0.5591 - val_loss: 0.6930
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5269 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6955
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 10ms/step - accuracy: 0.4775 - loss: 0.6938 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4397 - loss: 0.6942 - val_accuracy: 0.4409 -

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 21ms/step - accuracy: 0.4571 - loss: 0.6954 - val_accuracy: 0.4409 - val_loss: 0.6982
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4760 - loss: 0.6943 - val_accuracy: 0.5591 - val_loss: 0.6926
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5231 - loss: 0.6936 - val_accuracy: 0.4409 - val_loss: 0.6976
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4967 - loss: 0.6937 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5000 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6954
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4867 - loss: 0.6940 - val_accuracy: 0.5591 - val_loss: 0.6921
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4943 - loss: 0.6930 - val_accuracy: 0.4409 - val_loss: 0.6948
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5270 - loss: 0.6926 - val_accuracy: 0.4409 - 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 10ms/step - accuracy: 0.4871 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6935
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5099 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6936
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4999 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6941
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4993 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6939
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5052 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6945
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.5219 - loss: 0.6928 - val_accuracy: 0.4409 - val_loss: 0.6942
Model with 4 layers y 16 neurons -> Loss: 0.6933, accuracy: 0.4732
4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 30ms/step
ROC AUC: 0.49376399104573077

Entrenando modelo con 4 capas ocultas y 32 neuronas por capa


/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 4s 16ms/step - accuracy: 0.4857 - loss: 0.6937 - val_accuracy: 0.4409 - val_loss: 0.6932
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5326 - loss: 0.6944 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5220 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6939
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5117 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6948
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5074 - loss: 0.6934 - val_accuracy: 0.5591 - val_loss: 0.6928
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4879 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6933
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4771 - loss: 0.6933 - val_accuracy: 0.5591 - val_loss: 0.6908
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5032 - loss: 0.6931 - val_accuracy: 0.4409 - 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 2s 10ms/step - accuracy: 0.4507 - loss: 0.6977 - val_accuracy: 0.5433 - val_loss: 0.6932
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4561 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6935
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5010 - loss: 0.6935 - val_accuracy: 0.5591 - val_loss: 0.6928
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5304 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6952
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4887 - loss: 0.6939 - val_accuracy: 0.4409 - val_loss: 0.6943
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4969 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6953
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5159 - loss: 0.6929 - val_accuracy: 0.4409 - val_loss: 0.6953
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4894 - loss: 0.6937 - val_accuracy: 0.4409 - 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 19ms/step - accuracy: 0.5341 - loss: 0.6938 - val_accuracy: 0.4409 - val_loss: 0.7003
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 9ms/step - accuracy: 0.4848 - loss: 0.6959 - val_accuracy: 0.5591 - val_loss: 0.6929
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 9ms/step - accuracy: 0.5251 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 9ms/step - accuracy: 0.5078 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6946
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 8ms/step - accuracy: 0.5257 - loss: 0.6930 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.4794 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6939
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 12ms/step - accuracy: 0.4530 - loss: 0.6936 - val_accuracy: 0.4409 - val_loss: 0.6942
Model with 4 layers y 128 neurons -> Loss: 0.6931, accuracy: 0.5268
4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 20ms/step
ROC

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 11ms/step - accuracy: 0.5143 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5158 - loss: 0.6930 - val_accuracy: 0.4409 - val_loss: 0.6945
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5119 - loss: 0.6930 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4667 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6932
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4845 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6936
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5195 - loss: 0.6930 - val_accuracy: 0.4409 - val_loss: 0.6943
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5345 - loss: 0.6926 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5484 - loss: 0.6927 - val_accuracy: 0.4409 - 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 15ms/step - accuracy: 0.4617 - loss: 0.6951 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.4851 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6939
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5363 - loss: 0.6928 - val_accuracy: 0.4409 - val_loss: 0.6946
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 9ms/step - accuracy: 0.5264 - loss: 0.6928 - val_accuracy: 0.4409 - val_loss: 0.6945
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.5167 - loss: 0.6931 - val_accuracy: 0.5591 - val_loss: 0.6931
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.4532 - loss: 0.6932 - val_accuracy: 0.5591 - val_loss: 0.6927
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.4783 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6936
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5560 - loss: 0.6929 - val_accuracy: 0.4409 - 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 12ms/step - accuracy: 0.5065 - loss: 0.6939 - val_accuracy: 0.5591 - val_loss: 0.6925
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4780 - loss: 0.6935 - val_accuracy: 0.5591 - val_loss: 0.6931
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5272 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5150 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6944
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5288 - loss: 0.6927 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4883 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6939
Model with 6 layers y 64 neurons -> Loss: 0.6928, accuracy: 0.5268
4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 27ms/step
ROC AUC: 0.5062360089542692

Entrenando modelo con 6 capas ocultas y 128 neuronas por capa


/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 13ms/step - accuracy: 0.5039 - loss: 0.6953 - val_accuracy: 0.4409 - val_loss: 0.6936
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4759 - loss: 0.6935 - val_accuracy: 0.4409 - val_loss: 0.6933
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5303 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6942
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.4765 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6939
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5128 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6936
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5159 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4876 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6935
Model with 6 layers y 128 neurons -> Loss: 0.6932, accuracy: 0.4732
4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 26ms/step
ROC 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 6s 16ms/step - accuracy: 0.4888 - loss: 0.6939 - val_accuracy: 0.4409 - val_loss: 0.6952
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4689 - loss: 0.6938 - val_accuracy: 0.4409 - val_loss: 0.6934
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4702 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6934
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5409 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6942
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5130 - loss: 0.6930 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5214 - loss: 0.6930 - val_accuracy: 0.4409 - val_loss: 0.6939
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4872 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5118 - loss: 0.6931 - val_accuracy: 0.4409 - 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 12ms/step - accuracy: 0.4720 - loss: 0.6935 - val_accuracy: 0.5591 - val_loss: 0.6926
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5220 - loss: 0.6930 - val_accuracy: 0.5591 - val_loss: 0.6931
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.4916 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6935
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4873 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6933
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5035 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6933
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5116 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6934
Model with 8 layers y 32 neurons -> Loss: 0.6929, accuracy: 0.5268
4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 30ms/step
ROC AUC: 0.5060761112887752

Entrenando modelo con 8 capas ocultas y 64 neuronas por capa


/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 4s 20ms/step - accuracy: 0.4838 - loss: 0.6936 - val_accuracy: 0.4409 - val_loss: 0.6934
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 10ms/step - accuracy: 0.4824 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 9ms/step - accuracy: 0.5235 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 1s 10ms/step - accuracy: 0.4838 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6932
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - accuracy: 0.5100 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5200 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6938
Epoch 7/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.4944 - loss: 0.6932 - val_accuracy: 0.5591 - val_loss: 0.6931
Epoch 8/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.4923 - loss: 0.6932 - val_accuracy: 0.4409 

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 3s 14ms/step - accuracy: 0.4514 - loss: 0.6944 - val_accuracy: 0.5591 - val_loss: 0.6931
Epoch 2/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - accuracy: 0.5340 - loss: 0.6934 - val_accuracy: 0.4409 - val_loss: 0.6937
Epoch 3/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 5ms/step - accuracy: 0.5007 - loss: 0.6933 - val_accuracy: 0.4409 - val_loss: 0.6945
Epoch 4/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5106 - loss: 0.6931 - val_accuracy: 0.4409 - val_loss: 0.6941
Epoch 5/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.4326 - loss: 0.6938 - val_accuracy: 0.4409 - val_loss: 0.6933
Epoch 6/200
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - accuracy: 0.5079 - loss: 0.6932 - val_accuracy: 0.4409 - val_loss: 0.6942
Model with 8 layers y 128 neurons -> Loss: 0.6931, accuracy: 0.5268
4/4 ━━━━━━━━━━━━━━━━━━━━ 0s 30ms/step
ROC AUC: 0.49424368404221297


The best results were given when : Model with 8 layers y 64 neurons ->

*Loss: 0.6931,

*accuracy: 0.5268

*ROC AUC: 0.49600255836264784

## With Lag

In [38]:
df_1H['kk08_pos_score_lag1']=df_1H['kk08_pos_score'].shift(1)
df_1H['kk08_pos_score_lag2']=df_1H['kk08_pos_score'].shift(2)
df_1H['kk08_pos_score_lag3']=df_1H['kk08_pos_score'].shift(3)

In [45]:
df_x_2 = df_1H.dropna().reset_index(drop=True)[["kk08_pos_score", "kk08_pos_score_lag1", "kk08_pos_score_lag2", "kk08_pos_score_lag3"]]
df_y_2 = df_1H["Return_sign"].iloc[3:].reset_index(drop=True)

In [70]:
df_y_2.shape


(741,)

In [95]:
train_share = 0.85
df_x_2_train = df_x_2.iloc[:int(len(df_x_2)*train_share)].reset_index(drop=True)
df_y_2_train = df_y_2.iloc[:int(len(df_x_2)*train_share)].reset_index(drop=True)

df_x_2_test = df_x_2.iloc[int(len(df_x_2)*train_share):].reset_index(drop=True)
df_y_2_test = df_y_2.iloc[int(len(df_x_2)*train_share):].reset_index(drop=True)

In [96]:
train_share = 0.90
split_index = int(train_share * len(df_x_1_train))
X_2train, X_2valid = df_x_2_train[:split_index], df_x_2_train[split_index:]
y_2train, y_2valid = df_y_2_train[:split_index], df_y_2_train[split_index:]

In [64]:
X = df_x_2.fillna(0).to_numpy()  # Fills NaNs with 0 (or any other value)
# Or drop rows with NaN values:
X = df_x_2.dropna().to_numpy()

In [85]:
X.shape

(629, 4)

In [87]:
model_lag = Sequential([
    Dense(32, activation='relu', input_shape=(4,)),
    Dropout(0.3),
    Dense(16, activation='relu'),
    Dropout(0.3),
    Dense(8, activation='relu'),
    Dense(1, activation='sigmoid')
])

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


In [92]:
print("Shape of df_x_2_train:", df_x_2_train.shape)
print("Shape of df_y_2_train:", df_y_2_train.shape)

Shape of df_x_2_train: (629, 4)
Shape of df_y_2_train: (629,)


In [97]:
print("Shape of X_2valid:", X_2valid.shape)
print("Shape of y_2valid:", y_2valid.shape)


Shape of X_2valid: (61, 4)
Shape of y_2valid: (61,)


In [98]:
model_lag.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])

In [99]:
history = model_lag.fit(
    df_x_2_train, df_y_2_train,
    epochs=200,
    batch_size=32,
    validation_data=(X_2valid, y_2valid),
    callbacks=[early_stopping, model_checkpoint, TqdmCallback(verbose=1)],
    verbose=0
)

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

# Merge feelings and returns from Elkulako (Ignore)

In [17]:
df_elkulako = pd.read_csv('/content/sample_data/Scores_ElKulako (1).csv')

In [18]:
df_elkulako

0
0        {'label': 'Neutral', 'score': 0.7601109743118286}
1        {'label': 'Neutral', 'score': 0.7919839024543762}
2        {'label': 'Neutral', 'score': 0.5934157371520996}
3        {'label': 'Bullish', 'score': 0.5577491521835327}
4        {'label': 'Neutral', 'score': 0.6853246688842773}
...                                                    ...
1331663   {'label': 'Bullish', 'score': 0.591331958770752}
1331664  {'label': 'Neutral', 'score': 0.6384872794151306}
1331665  {'label': 'Neutral', 'score': 0.8317734003067017}
1331666  {'label': 'Neutral', 'score': 0.9091027975082397}
1331667   {'label': 'Neutral', 'score': 0.547605037689209}

[1331668 rows x 1 columns]

In [19]:
import ast


In [20]:
# Step 1: Convert the JSON-like strings into dictionaries
df_elkulako['0'] = df_elkulako['0'].apply(ast.literal_eval)

# Step 2: Extract 'label' and 'score' into new columns
df_elkulako['label'] = df_elkulako['0'].apply(lambda x: x['label'])
df_elkulako['score'] = df_elkulako['0'].apply(lambda x: x['score'])

In [21]:
df_elkulako


0    label     score
0        {'label': 'Neutral', 'score': 0.7601109743118286}  Neutral  0.760111
1        {'label': 'Neutral', 'score': 0.7919839024543762}  Neutral  0.791984
2        {'label': 'Neutral', 'score': 0.5934157371520996}  Neutral  0.593416
3        {'label': 'Bullish', 'score': 0.5577491521835327}  Bullish  0.557749
4        {'label': 'Neutral', 'score': 0.6853246688842773}  Neutral  0.685325
...                                                    ...      ...       ...
1331663   {'label': 'Bullish', 'score': 0.591331958770752}  Bullish  0.591332
1331664  {'label': 'Neutral', 'score': 0.6384872794151306}  Neutral  0.638487
1331665  {'label': 'Neutral', 'score': 0.8317734003067017}  Neutral  0.831773
1331666  {'label': 'Neutral', 'score': 0.9091027975082397}  Neutral  0.909103
1331667   {'label': 'Neutral', 'score': 0.547605037689209}  Neutral  0.547605

[1331668 rows x 3 columns]

In [22]:
df_elkulako['Positive_score'] = df_elkulako.apply(lambda row: 1 - row['score'] if row['label'] == 'Bearish' else row['score'], axis=1)


In [23]:
#Type of labels in column labels :
df_elkulako['label'].unique()

array(['Neutral', 'Bullish', 'Bearish'], dtype=object)

In [24]:
df_elkulako

0    label     score  \
0        {'label': 'Neutral', 'score': 0.7601109743118286}  Neutral  0.760111   
1        {'label': 'Neutral', 'score': 0.7919839024543762}  Neutral  0.791984   
2        {'label': 'Neutral', 'score': 0.5934157371520996}  Neutral  0.593416   
3        {'label': 'Bullish', 'score': 0.5577491521835327}  Bullish  0.557749   
4        {'label': 'Neutral', 'score': 0.6853246688842773}  Neutral  0.685325   
...                                                    ...      ...       ...   
1331663   {'label': 'Bullish', 'score': 0.591331958770752}  Bullish  0.591332   
1331664  {'label': 'Neutral', 'score': 0.6384872794151306}  Neutral  0.638487   
1331665  {'label': 'Neutral', 'score': 0.8317734003067017}  Neutral  0.831773   
1331666  {'label': 'Neutral', 'score': 0.9091027975082397}  Neutral  0.909103   
1331667   {'label': 'Neutral', 'score': 0.547605037689209}  Neutral  0.547605   

         Positive_score  
0              0.760111  
1              0.791984  
2              0.593416  
3              0.557749  
4              0.685325  
...                 ...  
1331663        0.591332  
1331664        0.638487  
1331665        0.831773  
1331666        0.909103  
1331667        0.547605  

[1331668 rows x 4 columns]

In [25]:
df_elkulako[df_elkulako['label']=="Bearish"]

0    label     score  \
6        {'label': 'Bearish', 'score': 0.6867599487304688}  Bearish  0.686760   
15       {'label': 'Bearish', 'score': 0.7277977466583252}  Bearish  0.727798   
20       {'label': 'Bearish', 'score': 0.9384021759033203}  Bearish  0.938402   
113        {'label': 'Bearish', 'score': 0.82611483335495}  Bearish  0.826115   
149      {'label': 'Bearish', 'score': 0.9663400053977966}  Bearish  0.966340   
...                                                    ...      ...       ...   
1331562  {'label': 'Bearish', 'score': 0.6561643481254578}  Bearish  0.656164   
1331577  {'label': 'Bearish', 'score': 0.9632375836372375}  Bearish  0.963238   
1331623  {'label': 'Bearish', 'score': 0.5134078860282898}  Bearish  0.513408   
1331656  {'label': 'Bearish', 'score': 0.5800412893295288}  Bearish  0.580041   
1331660  {'label': 'Bearish', 'score': 0.6792209148406982}  Bearish  0.679221   

         Positive_score  
6              0.313240  
15             0.272202  
20             0.061598  
113            0.173885  
149            0.033660  
...                 ...  
1331562        0.343836  
1331577        0.036762  
1331623        0.486592  
1331656        0.419959  
1331660        0.320779  

[87857 rows x 4 columns]

In [26]:
df_clean_tweets = pd.read_csv('/content/sample_data/Cleaned_tweets (1).csv')

In [44]:
current_df = pd.concat([df_clean_tweets,df_elkulako["Positive_score"]], axis=1)

In [45]:
current_df

datetime        date        username  \
0        2022-01-01 00:00:00+00:00  2022-01-01   CryptoNerdApp   
1        2022-01-01 17:10:02+00:00  2022-01-01   bitcoinalerts   
2        2022-01-01 17:10:00+00:00  2022-01-01    oneandonlypd   
3        2022-01-01 17:09:59+00:00  2022-01-01  CHAIRFORCE_BTC   
4        2022-01-01 17:09:58+00:00  2022-01-01  crypto_squared   
...                            ...         ...             ...   
1331663  2022-01-31 10:17:58+00:00  2022-01-31         cstross   
1331664  2022-01-31 10:18:02+00:00  2022-01-31       S_Edalati   
1331665  2022-01-31 10:18:05+00:00  2022-01-31  bitcoinpressuk   
1331666  2022-01-31 10:17:28+00:00  2022-01-31       AlfaHedge   
1331667  2022-01-31 23:59:57+00:00  2022-01-31         inanksm   

                                                      text  \
0        Current Price of Bitcoin:\n$46320 (-1.85%)\n\n...   
1        The year for Bitcoin — A 2021 roundup of the f...   
2        #Bitcoin is all our savings. https://t.co/SGgy...   
3        @gmekhail Isn't that usually shouted at the pe...   
4        What are Decentralized Apps or DApps?\n#Crypto...   
...                                                    ...   
1331663  @ignaziop1977 @MLK3030 @SHOKUNIN_STUDIO At the...   
1331664  Cryptocurrency Prices On January 31 2021: Know...   
1331665  #Bitcoin #BTC #CRYPTO US Financial Advisors Ex...   
1331666  How low can #Bitcoin go ???\n\nExperts weigh i...   
1331667  #Bitcoin  is one of the most important inventi...   

                       Date_hourly                   Date_min  \
0        2022-01-01 01:00:00+00:00  2022-01-01 00:01:00+00:00   
1        2022-01-01 18:00:00+00:00  2022-01-01 17:11:00+00:00   
2        2022-01-01 18:00:00+00:00  2022-01-01 17:11:00+00:00   
3        2022-01-01 18:00:00+00:00  2022-01-01 17:10:00+00:00   
4        2022-01-01 18:00:00+00:00  2022-01-01 17:10:00+00:00   
...                            ...                        ...   
1331663  2022-01-31 11:00:00+00:00  2022-01-31 10:18:00+00:00   
1331664  2022-01-31 11:00:00+00:00  2022-01-31 10:19:00+00:00   
1331665  2022-01-31 11:00:00+00:00  2022-01-31 10:19:00+00:00   
1331666  2022-01-31 11:00:00+00:00  2022-01-31 10:18:00+00:00   
1331667  2022-02-01 00:00:00+00:00  2022-02-01 00:00:00+00:00   

                          Date_day                   Date_30m  \
0        2022-01-02 00:00:00+00:00  2022-01-01 00:30:00+00:00   
1        2022-01-02 00:00:00+00:00  2022-01-01 17:30:00+00:00   
2        2022-01-02 00:00:00+00:00  2022-01-01 17:30:00+00:00   
3        2022-01-02 00:00:00+00:00  2022-01-01 17:30:00+00:00   
4        2022-01-02 00:00:00+00:00  2022-01-01 17:30:00+00:00   
...                            ...                        ...   
1331663  2022-02-01 00:00:00+00:00  2022-01-31 10:30:00+00:00   
1331664  2022-02-01 00:00:00+00:00  2022-01-31 10:30:00+00:00   
1331665  2022-02-01 00:00:00+00:00  2022-01-31 10:30:00+00:00   
1331666  2022-02-01 00:00:00+00:00  2022-01-31 10:30:00+00:00   
1331667  2022-02-01 00:00:00+00:00  2022-02-01 00:00:00+00:00   

                          Date_15m  \
0        2022-01-01 00:15:00+00:00   
1        2022-01-01 17:15:00+00:00   
2        2022-01-01 17:15:00+00:00   
3        2022-01-01 17:15:00+00:00   
4        2022-01-01 17:15:00+00:00   
...                            ...   
1331663  2022-01-31 10:30:00+00:00   
1331664  2022-01-31 10:30:00+00:00   
1331665  2022-01-31 10:30:00+00:00   
1331666  2022-01-31 10:30:00+00:00   
1331667  2022-02-01 00:00:00+00:00   

                                              cleaned_text  Positive_score  
0        current price bitcoin btc bitcoin cryptocurren...        0.760111  
1                     year bitcoin roundup flagship crypto        0.791984  
2                                           bitcoin saving        0.593416  
3        isnt usually shouted people promoting going bi...        0.557749  
4        decentralized apps dapps cryptonews bitcoin cr...        0.685325  
...    

In [46]:
current_df = current_df.sort_values(by="datetime").reset_index(drop=True)

In [47]:
current_df['datetime'] = pd.to_datetime(current_df['datetime']).dt.tz_localize(None)
current_df['Date_hourly'] = pd.to_datetime(current_df['Date_hourly']).dt.tz_localize(None)
current_df['Date_min'] = pd.to_datetime(current_df['Date_min']).dt.tz_localize(None)
current_df['Date_15m'] = pd.to_datetime(current_df['Date_15m']).dt.tz_localize(None)
current_df['Date_30m'] = pd.to_datetime(current_df['Date_30m']).dt.tz_localize(None)

In [31]:
current_df

datetime        date         username  \
0       2022-01-01 00:00:00  2022-01-01    CryptoNerdApp   
1       2022-01-01 00:00:00  2022-01-01      TrendSpider   
2       2022-01-01 00:00:00  2022-01-01  MadStudentScie1   
3       2022-01-01 00:00:01  2022-01-01       WIRTUALapp   
4       2022-01-01 00:00:01  2022-01-01     takeoff_tech   
...                     ...         ...              ...   
1331663 2022-01-31 23:59:46  2022-01-31      deyonte_btc   
1331664 2022-01-31 23:59:52  2022-01-31      Baripondiss   
1331665 2022-01-31 23:59:52  2022-01-31     galaxy_orion   
1331666 2022-01-31 23:59:55  2022-01-31  BitcoinFeesCash   
1331667 2022-01-31 23:59:57  2022-01-31          inanksm   

                                                      text  \
0        Current Price of Bitcoin:\n$46320 (-1.85%)\n\n...   
1        $BTC Continues to bounce off the YTD anchored ...   
2                 Generating misunderstanding for Bitcoin!   
3        HAPPY NEW YEAR 2022! Experience a new way to e...   
4        【$TKO token #1】\n\nTKO token will be a utility...   
...                                                    ...   
1331663  One of the largest asset managers on the plane...   
1331664  Lol... Oh, you just realized, you're still ear...   
1331665  @julesapril8 @jimcramer I know. \n\n- He instr...   
1331666  Updated Bitcoin transaction fees: \n    \nBCH ...   
1331667  #Bitcoin  is one of the most important inventi...   

                Date_hourly            Date_min                   Date_day  \
0       2022-01-01 01:00:00 2022-01-01 00:01:00  2022-01-02 00:00:00+00:00   
1       2022-01-01 01:00:00 2022-01-01 00:01:00  2022-01-02 00:00:00+00:00   
2       2022-01-01 01:00:00 2022-01-01 00:01:00  2022-01-02 00:00:00+00:00   
3       2022-01-01 01:00:00 2022-01-01 00:01:00  2022-01-02 00:00:00+00:00   
4       2022-01-01 01:00:00 2022-01-01 00:01:00  2022-01-02 00:00:00+00:00   
...                     ...                 ...                        ...   
1331663 2022-02-01 00:00:00 2022-02-01 00:00:00  2022-02-01 00:00:00+00:00   
1331664 2022-02-01 00:00:00 2022-02-01 00:00:00  2022-02-01 00:00:00+00:00   
1331665 2022-02-01 00:00:00 2022-02-01 00:00:00  2022-02-01 00:00:00+00:00   
1331666 2022-02-01 00:00:00 2022-02-01 00:00:00  2022-02-01 00:00:00+00:00   
1331667 2022-02-01 00:00:00 2022-02-01 00:00:00  2022-02-01 00:00:00+00:00   

                   Date_30m            Date_15m  \
0       2022-01-01 00:30:00 2022-01-01 00:15:00   
1       2022-01-01 00:30:00 2022-01-01 00:15:00   
2       2022-01-01 00:30:00 2022-01-01 00:15:00   
3       2022-01-01 00:30:00 2022-01-01 00:15:00   
4       2022-01-01 00:30:00 2022-01-01 00:15:00   
...                     ...                 ...   
1331663 2022-02-01 00:00:00 2022-02-01 00:00:00   
1331664 2022-02-01 00:00:00 2022-02-01 00:00:00   
1331665 2022-02-01 00:00:00 2022-02-01 00:00:00   
1331666 2022-02-01 00:00:00 2022-02-01 00:00:00   
1331667 2022-02-01 00:00:00 2022-02-01 00:00:00   

                                              cleaned_text  Positive_score  
0        current price bitcoin btc bitcoin cryptocurren...        0.760111  
1           btc continues bounce ytd anchored vwap bitcoin        0.923622  
2                      generating misunderstanding bitcoin        0.252997  
3        happy new year experience new way exercise wir...        0.513676  
4        tko token tko token utility token service fee ...        0.520613  
...                                                    ...             ...  
1331663  one largest asset manager planet trillion aum ...        0.651603  
1331664  lol oh realized youre still early welcome bitcoin        0.765540  
1331665  know instruct agency ban bitcoin russia get wo...        0.042174  
1331666  updated bitcoin transaction fee bch next block...        0.655889  
1331667  bitcoin one important invention human history ...        0.547605  

[1331668 rows x 11 columns]

In [43]:
current_df.to_csv("Tweets_with_sentiment_elkulako.csv", index = False)

In [33]:
#Group the dataset by 15 min hour and 30 min
df_15m = current_df.groupby("Date_15m").agg({"datetime":list,"text":"\n".join,"username":"\n".join,"Positive_score":"mean"}).reset_index()
df_30m = current_df.groupby("Date_30m").agg({"datetime":list,"text":"\n".join,"username":"\n".join,"Positive_score":"mean"}).reset_index()
df_1H = current_df.groupby("Date_hourly").agg({"datetime":list,"text":"\n".join,"username":"\n".join,"Positive_score":"mean"}).reset_index()

In [34]:
sentiment_15m = df_15m

In [35]:
fin_15m = pd.read_csv("/content/sample_data/BTCUSDT_15m.csv")

In [36]:
fin_15m

date      open     close    Return  Return_sign
0     2022-01-01 00:00:00  46216.93  46332.51  0.002501            1
1     2022-01-01 00:15:00  46332.52  46375.42  0.000926            1
2     2022-01-01 00:30:00  46375.42  46610.81  0.005076            1
3     2022-01-01 00:45:00  46610.81  46656.13  0.000972            1
4     2022-01-01 01:00:00  46656.14  46766.99  0.002376            1
...                   ...       ...       ...       ...          ...
2971  2022-01-31 22:45:00  38590.00  38412.46 -0.004601            0
2972  2022-01-31 23:00:00  38412.45  38451.36  0.001013            1
2973  2022-01-31 23:15:00  38451.36  38412.23 -0.001018            0
2974  2022-01-31 23:30:00  38412.23  38369.11 -0.001123            0
2975  2022-01-31 23:45:00  38369.12  38466.90  0.002548            1

[2976 rows x 5 columns]

In [37]:
# Convert both columns to datetime format
sentiment_15m["Date_15m"] = pd.to_datetime(sentiment_15m["Date_15m"])
fin_15m["date"] = pd.to_datetime(fin_15m["date"])

# Now perform the merge
df_15m = pd.merge(sentiment_15m, fin_15m, left_on="Date_15m", right_on="date", how="inner")


In [39]:
df_15m
# Remove colums datetime	text	username
df_15m = df_15m.drop(columns=["datetime","text","username","date"])

In [40]:
df_15m

Date_15m  Positive_score      open     close    Return  \
0    2022-01-01 00:15:00        0.639253  46332.52  46375.42  0.000926   
1    2022-01-01 00:30:00        0.647286  46375.42  46610.81  0.005076   
2    2022-01-01 00:45:00        0.662242  46610.81  46656.13  0.000972   
3    2022-01-01 01:00:00        0.623321  46656.14  46766.99  0.002376   
4    2022-01-01 01:15:00        0.649561  46766.99  46815.33  0.001034   
...                  ...             ...       ...       ...       ...   
2970 2022-01-31 22:45:00        0.649623  38590.00  38412.46 -0.004601   
2971 2022-01-31 23:00:00        0.658045  38412.45  38451.36  0.001013   
2972 2022-01-31 23:15:00        0.656948  38451.36  38412.23 -0.001018   
2973 2022-01-31 23:30:00        0.646117  38412.23  38369.11 -0.001123   
2974 2022-01-31 23:45:00        0.656399  38369.12  38466.90  0.002548   

      Return_sign  
0               1  
1               1  
2               1  
3               1  
4               1  
...           ...  
2970            0  
2971            1  
2972            0  
2973            0  
2974            1  

[2975 rows x 6 columns]

In [41]:
def create_model(num_layers=6, neurons_per_layer=48, input_shape=(10,), output_units=1, activation='relu'):
    model = Sequential()
    model.add(Dense(neurons_per_layer, activation=activation, input_shape=input_shape))

    # Agregar las capas ocultas especificadas
    for _ in range(num_layers - 1):
        model.add(Dense(neurons_per_layer, activation=activation))

    # Capa de salida
    model.add(Dense(output_units, activation='sigmoid'))  # Cambia 'sigmoid' a 'softmax' si tienes una clasificación multiclase

    # Compilar el modelo
    model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
    return model

X  = np.array(df_15m['Positive_score']).reshape(-1, 1)
y  = np.array(df_15m['Return_sign'])
X_train, X_test, Y_train, Y_test = train_test_split(X, y, test_size=0.2, random_state=42)


# Definir parámetros de la red
num_layers_list = [2, 3, 4,6,8]  # Diferentes números de capas a probar
neurons_per_layer_list = [16, 32, 64,128]  # Diferentes números de neuronas en cada capa

# Probar diferentes configuraciones de capas y neuronas
for num_layers in num_layers_list:
    for neurons_per_layer in neurons_per_layer_list:
        print(f"\nEntrenando modelo con {num_layers} capas ocultas y {neurons_per_layer} neuronas por capa")

        # Crear el modelo
        model = create_model(num_layers=num_layers, neurons_per_layer=neurons_per_layer, input_shape=(1,))

        # Definir early stopping para evitar el sobreajuste
        early_stopping = EarlyStopping(monitor='val_loss', patience=5, restore_best_weights=True)

        # Entrenar el modelo
        history = model.fit(X_train, Y_train, validation_split=0.2, epochs=200, batch_size=16, callbacks=[early_stopping,TqdmCallback(verbose=1)], verbose=1)

        # Evaluar el modelo
        loss, accuracy = model.evaluate(X_test, Y_test, verbose=0)
        print(f"Modelo con {num_layers} capas y {neurons_per_layer} neuronas -> Pérdida: {loss:.4f}, Precisión: {accuracy:.4f}")

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:87: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)



Entrenando modelo con 2 capas ocultas y 16 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.5035 - loss: 0.6951 - val_accuracy: 0.5021 - val_loss: 0.6937
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.4938 - loss: 0.6939 - val_accuracy: 0.5021 - val_loss: 0.6938
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.5223 - loss: 0.6922 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5284 - loss: 0.6923 - val_accuracy: 0.5021 - val_loss: 0.6931
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.4962 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6931
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.5191 - loss: 0.6928 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.4765 - loss: 0.6937 - val_accuracy: 0.5021 - val_loss: 0.6938
Epoch 8/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5226 - loss: 0.6923 - val_accu

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.4876 - loss: 0.6938 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5040 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6936
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.4896 - loss: 0.6941 - val_accuracy: 0.5021 - val_loss: 0.6935
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5037 - loss: 0.6936 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5161 - loss: 0.6928 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.4868 - loss: 0.6941 - val_accuracy: 0.5021 - val_loss: 0.6933
Modelo con 2 capas y 32 neuronas -> Pérdida: 0.6923, Precisión: 0.5328

Entrenando modelo con 2 capas ocultas y 64 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 3s 7ms/step - accuracy: 0.4900 - loss: 0.6940 - val_accuracy: 0.4979 - val_loss: 0.6931
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5004 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6934
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4990 - loss: 0.6936 - val_accuracy: 0.5021 - val_loss: 0.6975
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.4974 - loss: 0.6980 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.5179 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.5152 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6933
Modelo con 2 capas y 64 neuronas -> Pérdida: 0.6932, Precisión: 0.4672

Entrenando modelo con 2 capas ocultas y 128 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.5114 - loss: 0.6947 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.5070 - loss: 0.6947 - val_accuracy: 0.5021 - val_loss: 0.6950
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.5193 - loss: 0.6928 - val_accuracy: 0.4979 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.5022 - loss: 0.6936 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.5357 - loss: 0.6919 - val_accuracy: 0.4979 - val_loss: 0.6936
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.4927 - loss: 0.6938 - val_accuracy: 0.5021 - val_loss: 0.6935
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5121 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 8/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.4828 - loss: 0.6936 - val_accu

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.5058 - loss: 0.6951 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.4822 - loss: 0.6938 - val_accuracy: 0.5021 - val_loss: 0.6937
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.5132 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5114 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.4996 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5127 - loss: 0.6929 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5129 - loss: 0.6928 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 8/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5154 - loss: 0.6929 - val_accu

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 3s 8ms/step - accuracy: 0.4984 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.5038 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6934
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5194 - loss: 0.6927 - val_accuracy: 0.5021 - val_loss: 0.6934
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5202 - loss: 0.6927 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5121 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6934
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5120 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.4961 - loss: 0.6936 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 8/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.4804 - loss: 0.6938 - val_accu

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.4861 - loss: 0.6936 - val_accuracy: 0.5021 - val_loss: 0.6935
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5182 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6938
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.5249 - loss: 0.6925 - val_accuracy: 0.4979 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5029 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6931
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.4832 - loss: 0.6934 - val_accuracy: 0.5021 - val_loss: 0.6935
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5136 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.5069 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 8/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.4917 - loss: 0.6939 - val_accu

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 7ms/step - accuracy: 0.5074 - loss: 0.6941 - val_accuracy: 0.4979 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5013 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6935
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.4774 - loss: 0.6945 - val_accuracy: 0.5021 - val_loss: 0.6940
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.4937 - loss: 0.6946 - val_accuracy: 0.5021 - val_loss: 0.6937
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5032 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5092 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6935
Modelo con 3 capas y 128 neuronas -> Pérdida: 0.6937, Precisión: 0.4672

Entrenando modelo con 4 capas ocultas y 16 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 3s 8ms/step - accuracy: 0.5140 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.4672 - loss: 0.6938 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4969 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5017 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5068 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6934
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.4876 - loss: 0.6945 - val_accuracy: 0.5021 - val_loss: 0.6935
Modelo con 4 capas y 16 neuronas -> Pérdida: 0.6922, Precisión: 0.5328

Entrenando modelo con 4 capas ocultas y 32 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.4804 - loss: 0.6950 - val_accuracy: 0.5021 - val_loss: 0.6931
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5142 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.5032 - loss: 0.6934 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5319 - loss: 0.6927 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step - accuracy: 0.4992 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5120 - loss: 0.6930 - val_accuracy: 0.5021 - val_loss: 0.6933
Modelo con 4 capas y 32 neuronas -> Pérdida: 0.6927, Precisión: 0.5328

Entrenando modelo con 4 capas ocultas y 64 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 4ms/step - accuracy: 0.5227 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6931
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5049 - loss: 0.6938 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5169 - loss: 0.6928 - val_accuracy: 0.4979 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.4955 - loss: 0.6938 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5223 - loss: 0.6929 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5027 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6933
Modelo con 4 capas y 64 neuronas -> Pérdida: 0.6929, Precisión: 0.5328

Entrenando modelo con 4 capas ocultas y 128 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 5s 10ms/step - accuracy: 0.4766 - loss: 0.6937 - val_accuracy: 0.4979 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.4967 - loss: 0.6930 - val_accuracy: 0.4979 - val_loss: 0.6935
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4961 - loss: 0.6941 - val_accuracy: 0.5021 - val_loss: 0.6931
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5165 - loss: 0.6929 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.4942 - loss: 0.6938 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5037 - loss: 0.6934 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.5062 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 8/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5201 - loss: 0.6926 - val_acc

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 4s 5ms/step - accuracy: 0.5031 - loss: 0.6934 - val_accuracy: 0.5021 - val_loss: 0.6931
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5191 - loss: 0.6930 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5069 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5023 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5033 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5136 - loss: 0.6929 - val_accuracy: 0.5021 - val_loss: 0.6933
Modelo con 6 capas y 16 neuronas -> Pérdida: 0.6927, Precisión: 0.5328

Entrenando modelo con 6 capas ocultas y 32 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 3s 5ms/step - accuracy: 0.4847 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6935
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5092 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5269 - loss: 0.6923 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5037 - loss: 0.6934 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5003 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5106 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5029 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 8/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5338 - loss: 0.6919 - val_accu

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 4s 8ms/step - accuracy: 0.4919 - loss: 0.6945 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4866 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5162 - loss: 0.6929 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4967 - loss: 0.6936 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4975 - loss: 0.6937 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.4864 - loss: 0.6938 - val_accuracy: 0.5021 - val_loss: 0.6936
Modelo con 6 capas y 64 neuronas -> Pérdida: 0.6925, Precisión: 0.5328

Entrenando modelo con 6 capas ocultas y 128 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 5s 9ms/step - accuracy: 0.5149 - loss: 0.6936 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.4935 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 9ms/step - accuracy: 0.5185 - loss: 0.6930 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 10ms/step - accuracy: 0.5316 - loss: 0.6923 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 8ms/step - accuracy: 0.5036 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 2s 13ms/step - accuracy: 0.5041 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6932
Modelo con 6 capas y 128 neuronas -> Pérdida: 0.6925, Precisión: 0.5328

Entrenando modelo con 8 capas ocultas y 16 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 7s 9ms/step - accuracy: 0.4915 - loss: 0.6934 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5102 - loss: 0.6930 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.4991 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5050 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5084 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 7ms/step - accuracy: 0.4987 - loss: 0.6934 - val_accuracy: 0.5021 - val_loss: 0.6933
Modelo con 8 capas y 16 neuronas -> Pérdida: 0.6925, Precisión: 0.5328

Entrenando modelo con 8 capas ocultas y 32 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 4s 5ms/step - accuracy: 0.5110 - loss: 0.6935 - val_accuracy: 0.5021 - val_loss: 0.6935
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.4644 - loss: 0.6949 - val_accuracy: 0.5021 - val_loss: 0.6937
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.4918 - loss: 0.6946 - val_accuracy: 0.5021 - val_loss: 0.6936
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5139 - loss: 0.6929 - val_accuracy: 0.5021 - val_loss: 0.6934
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5210 - loss: 0.6925 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step - accuracy: 0.5070 - loss: 0.6932 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5121 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 8/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 3ms/step - accuracy: 0.5039 - loss: 0.6933 - val_accu

0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 5s 10ms/step - accuracy: 0.5042 - loss: 0.6933 - val_accuracy: 0.5021 - val_loss: 0.6931
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.4918 - loss: 0.6934 - val_accuracy: 0.5021 - val_loss: 0.6934
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5189 - loss: 0.6929 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 9ms/step - accuracy: 0.5299 - loss: 0.6926 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - accuracy: 0.5149 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 8ms/step - accuracy: 0.5104 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6934
Modelo con 8 capas y 64 neuronas -> Pérdida: 0.6929, Precisión: 0.5328

Entrenando modelo con 8 capas ocultas y 128 neuronas por capa


0epoch [00:00, ?epoch/s]

0batch [00:00, ?batch/s]

Epoch 1/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 4s 6ms/step - accuracy: 0.4618 - loss: 0.6943 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 2/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5298 - loss: 0.6927 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 3/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.4584 - loss: 0.6937 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 4/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5167 - loss: 0.6928 - val_accuracy: 0.5021 - val_loss: 0.6932
Epoch 5/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - accuracy: 0.5175 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6934
Epoch 6/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 4ms/step - accuracy: 0.5129 - loss: 0.6930 - val_accuracy: 0.5021 - val_loss: 0.6933
Epoch 7/200
119/119 ━━━━━━━━━━━━━━━━━━━━ 1s 5ms/step - accuracy: 0.5075 - loss: 0.6931 - val_accuracy: 0.5021 - val_loss: 0.6934
Modelo con 8 capas y 128 neuronas -> Pérdida: 0.6925, Precisión: 0.5328


In [49]:
twitter_15min

datetime  aggregated_positive_score
0    2022-01-01 00:00:00                   0.639253
1    2022-01-01 00:15:00                   0.647286
2    2022-01-01 00:30:00                   0.662242
3    2022-01-01 00:45:00                   0.623321
4    2022-01-01 01:00:00                   0.649561
...                  ...                        ...
2971 2022-01-31 22:45:00                   0.658045
2972 2022-01-31 23:00:00                   0.656948
2973 2022-01-31 23:15:00                   0.646117
2974 2022-01-31 23:30:00                   0.656399
2975 2022-01-31 23:45:00                   0.660316

[2976 rows x 2 columns]

In [ ]:
# Asegúrate de que las columnas de fecha estén en formato datetime
current_df['datetime'] = pd.to_datetime(current_df['datetime'])
fin_15m['date'] = pd.to_datetime(fin_15m['date'])
# Configura el índice para resampling basado en datetime
current_df.set_index('datetime', inplace=True)
